rust/processor/src/grpc_stream.rs

```diff
@@ -99,7 +100,13 @@ pub async fn get_stream(
         end_version = ending_version,
         "[Parser] Setting up GRPC client"
     );
-    let mut rpc_client = match RawDataClient::connect(channel).await {
+
+    // Use tokio::time::timeout to set a timeout for the request
+    let connect_res = timeout(Duration::from_secs(1), RawDataClient::connect(channel))
```

Comment:
If it times out, can we add some retry logic?

Reply:
Yeee I can add three retries